### PR TITLE
rename iso_graphical to iso_plasma5

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
In the upcoming possibility of there being an `iso_graphical_gnome3` renaming `iso_graphical`
to `iso_graphical_plasma5` is needed because there could be multiple `iso_graphical`'s.

I've been informed that changing this will require updating the channel generation script and website.
Please merge with
* [ ] https://github.com/NixOS/nixos-channel-scripts/pull/25
* [ ] https://github.com/NixOS/nixos-homepage/pull/297

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by samueldr — APPROVED** [review]
:+1: for the more specific name.

**Comment by edolstra** [comment]
Not in favor of this because it's sufficient to provide 1 graphical ISO and this would open the floodgates to ISOs for every desktop environment out there. Also having multiple ISOs makes testing harder.

**Comment by edolstra** [comment]
So it would be interesting to know what the size of a combined Plasma + Gnome ISO is. Presumably it's quite a bit less than the separate ISOs.

**Comment by edolstra** [comment]
Maybe ISO (and other image) generation should be moved out of the main jobsets into a jobset that runs at a lower frequency. After all, channel users don't really care whether the ISOs built correctly, so ISOs don't need to be a channel blocker.

**Comment by worldofpeace** [comment]
> Maybe ISO (and other image) generation should be moved out of the main jobsets into a jobset that runs at a lower frequency. After all, channel users don't really care whether the ISOs built correctly, so ISOs don't need to be a channel blocker.

That sounds perfect :sparkles:

**Comment by aanderse** [comment]
@edolstra while I definitely don't think NixOS should have an ISO for every single desktop environment out there... you had made the valid counterpoint to this (during #officehours) that gnome and kde are **the** primary desktop environments, and have been for many years now... so it isn't hard to draw the line at kde **plus** gnome, with no more ISOs after that.

Let us keep in mind that we are all very technical people who know how to work with NixOS to get exactly what we want, but our knowledge and ability does not represent the majority of nix users, nor does it represent people who we want to become NixOS users. When people in the community put an effort forward to improve the UX, especially for new users, I think we should see what we can do to encourage and help that. That being said, let me know if there is any help I can provide you with on this effort @worldofpeace! Thanks for thinking about new and potential users and the positive experience we want them to have :heart:

**Comment by worldofpeace** [comment]
Thank you @aanderse :sparkles: I really do appreciate your perspective.

---

@edolstra 

##### The following is just the general thoughts floating around in my head

I also want to note that someone is or should be responsible that GNOME in nixpkgs is suitable for production on users machines, and honestly has been part of the Release Engineering process.

I'd say from my experience improving this and user feedback it's extremely suitable for an ISO in distribution, I'm already responsible for GNOME's user experience. If we need policies for what desktop environments should be marked as officially supported in nixpkgs we should have that. I'd suggest a form of a group, comprised of
* People responsible for QA
* Release Engineering
* Infrastructure
* Desktop environment maintainers

And people who can be responsible for the final say like you.
This would create a great bridge of communication.

I'd also like to address the concern of people not knowing which ISO to choose.
Someone would already know this by which desktop environment they were going to use, and telling if something is better is difficult. NixOS does not have a default desktop environment, it's flexible for anyone's preferences and they can simply declare what it is they want with an option.

This is nice but adds a pretty serious level of ambiguity to users, this is why Fedora for example has "Spins" and Ubuntu's default is now GNOME but there's still things like Kubuntu or Ubuntu Mate.
Without any level of definition, to start with an ISO, I can't communicate a user experience that could be pleasant to users because they simply cannot see it.

**Comment by samueldr** [comment]
A random though that occurred: using a graphical iso (or session) should pre-seed the configuration.nix with the configuration for that DE. This would add value to having the additional DE, as this allows the end-user to test the DE, and have a reasonable assumption that once it is installed it will run as it did with the installer media. Though, this does imply more work to get there.

**Comment by worldofpeace** [comment]
> A random though that occurred: using a graphical iso (or session) should pre-seed the configuration.nix with the configuration for that DE. This would add value to having the additional DE, as this allows the end-user to test the DE, and have a reasonable assumption that once it is installed it will run as it did with the installer media.

That's an excellent idea, and intuitive.

And it looks like there might be a way put this together because of https://github.com/NixOS/nixpkgs/commit/810388afd21154aa99c3bdabd7bc2fe04150ed92. Or at least clear how to add this.

You'd have a configuration seed/switch for
* iso_minimal
* iso_graphical_plasma5
* iso_graphical_gnome3
* sd_image?

A followup to definitely consider.

**Comment by worldofpeace** [comment]
So, two months later and a nixos release... I think we could go for this in 20.03, and as stated, I'm completely willing do whatever I can to have this within my abilities. And of course with the co-operation of my peers.

**Comment by davidak** [comment]
I like the direction this discussion is going!

Some thoughts:

Why not call the ISOs `iso_graphical_kde` and `iso_graphical_gnome`? I think "plasma5" is more an implementation detail the end-user don't need to know and KDE is a more common term (see other distros iso name). We will most likely not have "iso_graphical_gnome2" and "iso_graphical_gnome3" at the same time, so the explicit version number is not needed there and is more confusing for users.

**Comment by worldofpeace** [comment]
> I like the direction this discussion is going!
> 
> Some thoughts:
> 
> Why not call the ISOs `iso_graphical_kde` and `iso_graphical_gnome`? I think "plasma5" is more an implementation detail the end-user don't need to know and KDE is a more common term (see other distros iso name). We will most likely not have "iso_graphical_gnome2" and "iso_graphical_gnome3" at the same time, so the explicit version number is not needed there and is more confusing for users.

I kinda agree, but the nixos options are
* `services.xserver.desktopManager.plasma5.enable`
* `services.xserver.desktopManager.gnome3.enable`

And that particular nixos configuration is what differentiates these artifacts, so I think it's best we name them by this.

**Review by davidak — APPROVED** [review]
LGTM

File: nixos/release.nix
Comment (edolstra) [inline nixos/release.nix]: For brevity, maybe call it `iso_plasma5`? "Graphical" is implied by Plasma.

Comment (worldofpeace) [reply nixos/release.nix]: Yeah that is better :+1:

**Pull request merged**
